src/trusted/service_runtime/nacl_syscall_common.c

```diff
@@ -3137,6 +3137,55 @@ int32_t NaClSysImcSocketPair(struct NaClAppThread *natp,
   return retval;
 }
 
+int32_t NaClSysSocketPair(struct NaClAppThread *natp,
+                          int                  domain,
+                          int                  type,
+                          int                  protocol,
+                          int                  *fds) {
+
+  struct NaClApp          *nap = natp->nap;
+  void                    *xchangedata = NULL;
+  int32_t                 retval;
+
+  NaClLog(2, "Cage %d Entered NaClSysSocketPair(0x%08"NACL_PRIxPTR", "
+           "%d, %d, %d, 0x%08"NACL_PRIx32")\n",
+           nap->cage_id, (uintptr_t)natp, domain, type, protocol, (uintptr_t)fds);
+
+  //Preprocessing start
+  xchangedata = malloc(sizeof(struct NaClHostDesc)*2);
```

Comment:
This whole function could be much neater. I know its just basically copying the pre/post processes around the RPC call, but it would be better if you organized it better. Using a more descriptive variable than "xchangedata" and things like moving the declarations to the beginning would be a good start.

Reply:
I agree